File: tests/integration/features/type-vocabulary/type-vocabulary.ts
// Principle #14: type references wrap their name in a <name> child.
// Every <type> element should contain <name>X</name>, not bare text.
// Covers: plain types, generic types, base-class/implements, function types.

type Id = number;
type Handler = (x: number) => void;
type Box<T> = Array<T>;

class Animal {}
interface Barker { bark(): void; }
class Dog extends Animal implements Barker {
    bark(): void {}
}

File: tests/integration/features/type-vocabulary/type-vocabulary.cs
// Principle #14: type references wrap their name in a <name> child.
// Covers: plain types, generic types, base list (extends + implements),
// type parameter declarations.

using System.Collections.Generic;

interface IBarker { void Bark(); }
class Animal {}

class Dog<T> : Animal, IBarker where T : Animal
{
    public T Owner;
    public List<string> Tags;
    public void Bark() {}
}
